cisagov/crossfeed
# Add Permission Checking for React Routes

#  #

## 🗣 Description ##
closes #2472 and addresses the issue that allows users with incorrect permission types to view pages outside of the intended permission level. Modified the route definitions to accept an array of allowed permission levels. Any user that doesn't have the set permissions, will be logged out and redirected to the login page.




## 💭 Motivation and context ##
Addressing security concerns on frontend application routes.






## 🧪 Testing ##
Attempt to access any of the frontend routes listed below with and without a user with the below "userType" values 
("standard", "globalView", "globalAdmin", or "regionalAdmin") and confirm that the user can either access the page or is logged out and redirected to the login screen as expected. The userType and permission "globalAdmin" has access to all existing routes and is assumed in the" permissions" props without specific assignment in the RouteGuard components.

The current requested permission assignments reflect the following:

![image](https://github.com/cisagov/crossfeed/assets/6486652/2ca183af-b6c6-480b-81b0-25ea9ce19ba7)

Example of a local test would be the following process:

1. Checkout this branch locally in Crossfeed project.
2. Pull/Rebuild the project.
3. Verify that the test user is assigned a "userType" of either "regionalAdmin" or "globalAdmin" _(Matches permission image reference)._
4. Login to Crossfeed with the test user.
5. Attempt to access the Region Admin Dashboord page at [https://localhost/region-admin-dasboard](https://localhost/region-admin-dasboard)
6. . Confirm that the user can access the page if the logged in user **HAS** a "userType" of either "regionalAdmin" or "globalAdmin".
7. Confirm that the user is logged out and redirected to login screen if user **DOES NOT HAVE** a "userType" of either "regionalAdmin" or "globalAdmin".









## ✅ Pre-approval checklist ##






      in code comments.
      to reflect the changes in this PR.
